host/optee/linux/enclave.c

```diff
@@ -524,6 +569,9 @@ oe_result_t oe_terminate_enclave(oe_enclave_t* enclave)
     /* Finalize the context against OP-TEE */
     TEEC_FinalizeContext(&enclave->ctx);
 
+    /* Destroy the ecall id table */
+    free(enclave->ecall_id_table);
```

Comment:
Should we check if pointer is NULL? Not sure if that's even possible tho.

Reply:
free(NULL) is valid, and a noop. I have added an if check to be consistent with other uses of free.

Reply:
Should not have an if check, it just clutters up the code for no value, and (unless there is cross-function optimization) results in increased binary size and instruction branching.